src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -150,7 +150,16 @@ def _crawl(self) -> list[GlobalInitScriptInfo]:
 
     def _assess_global_init_scripts(self, all_global_init_scripts):
         for gis in all_global_init_scripts:
-            global_init_script_info = GlobalInitScriptInfo(gis.script_id, gis.name, gis.created_by, gis.enabled, 1, "")
+            if not gis.created_by:
+                logger.warning(
+                    f"Script {gis.name} have Unknown creator, it means that the original creator has been deleted"
+                    f" and should be re-created"
+                )
+                global_init_script_info = GlobalInitScriptInfo(gis.script_id, gis.name, "", gis.enabled, 1, "")
```

Comment:
Pattern for entire PR: don't surround creation of entire object with "if", but rather surround introduce a variable for the username.

Having two code paths for append may cause difficult bugs in the future